src/emu/ioport.cpp

```diff
@@ -235,6 +235,7 @@ const struct
 	{ INPUT_STRING_Reverse, "Reverse" },
 	{ INPUT_STRING_Alternate, "Alternate" },
 	{ INPUT_STRING_None, "None" },
+	{ INPUT_STRING_Diagnostic, "Diagnostic" },
```

Comment:
Please don’t add an input type for this.  Just use `PORT_CONFNAME` etc or `PORT_BIT( ..., ..., IPT_OTHER ) PORT_TOGGLE PORT_NAME(...)`

Reply:
Aha! I only did it this way because I hadn't figured that out. :)